app/models/reviewable_score.rb

```diff
@@ -11,20 +13,28 @@ class ReviewableScore < ActiveRecord::Base
   # To keep things simple the types correspond to `PostActionType` for backwards
   # compatibility, but we can add extra reasons for scores.
   def self.types
-    @types ||= PostActionType.flag_types.merge(PostActionType.score_types)
+    Discourse
+      .cache
+      .fetch(REVIEWABLE_SCORE_TYPES_KEY) do
+        score_types = PostActionType.flag_types.merge(PostActionType.score_types)
+        score_types.merge!(@plugin_types) if @plugin_types
+        score_types
+      end
   end
 
   # When extending post action flags, we need to call this method in order to
   # get the latests flags.
   def self.reload_types
-    @types = nil
+    Discourse.cache.delete(REVIEWABLE_SCORE_TYPES_KEY)
     types
   end
 
   def self.add_new_types(type_names)
     next_id = types.values.max + 1
+    @plugin_types ||= {}
```

Comment:
This doesn't seem to be multisite safe considering that it is based off `types` which is multisite safe.